src/ngraph/node.hpp

```diff
@@ -106,18 +107,33 @@ namespace ngraph
 
         Node(const std::string& node_type, const NodeVector& arguments, size_t output_size = 1);
 
+        /// \brief Construct an unitialized Node
+        Node() {}
+        /// Constructor for Node subclasses that have metaclasses.
+        /// \param arguments The 0th output of node i will connect to input i
+        /// \param output_size Number of outputs for this node
+        Node(const NodeVector& arguments, size_t output_size = 1);
+
         virtual void generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector& deltas) {}
     public:
         virtual ~Node();
+
+        /// Sets/replaces the arguments with new arguments.
+        void set_arguments(const NodeVector& arguments);
+        void set_arguments(const OutputVector& arguments);
+        void set_argument(size_t position, const Output<Node>& argument);
+
+        /// Sets the number of outputs
+        void set_output_size(size_t output_size);
+
         void revalidate_and_infer_types() { validate_and_infer_types(); }
         // Called after transition
         void delayed_validate_and_infer_types();
 
         /// \brief Get the string name for the type of the node, such as `Add` or `Multiply`.
         ///        The class name, must not contain spaces as it is used for codegen.
         /// \returns A const reference to the node's type name
-        const std::string& description() const;
-
+        virtual const std::string& description() const;
```

Comment:
Will this be de-virtualized once all ops are on the new metaclass regime?

Reply:
So far, the vtbl has been able to do everything a metaclass would do.